common/http.go

```diff
@@ -0,0 +1,79 @@
+package common
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"gopkg.in/go-playground/validator.v9"
+
+	"github.com/pkg/errors"
+)
+
+type ErrorResponse struct {
+	Success bool `json:"success,omitempty"`
+
+	// Error is the error message if Success is false
+	Error string `json:"error,omitempty"`
+
+	// Code is set if Success is false
+	Code int `json:"code,omitempty"`
+}
+
+var errNilBody = errors.Errorf("expecting a non-nil body")
+
+// ParseRequestJSON unmarshals into save, the body of the
+// request. It closes the body of the request after parsing.
+// Since it uses json.Unmarshal, save must be of a pointer type
+// or compatible with json.Unmarshal.
+func ParseRequestJSON(r *http.Request, save interface{}) error {
```

Comment:
Thank you for the request for updates. Sure, I can change it and that might be a less broader API for the io.Reader instead of a io.ReadCloser or *http.Request. Coming up in a few.